roles/openshift_autoheal/files/template.yml

```diff
@@ -0,0 +1,198 @@
+  roleRef:
+    kind: ClusterRole
+    name: system:auth-delegator
+  subjects:
+  - kind: ServiceAccount
+    namespace: openshift-autoheal
+    name: autoheal
+
+- apiVersion: v1
+  kind: Secret
+  metadata:
+    name: autoheal-config
+    labels:
+      app: autoheal
+  data:
+    autoheal.yml: ${CONFIG}
+
+- apiVersion: v1
+  kind: Secret
+  metadata:
+    name: autoheal-proxy-cookie
+  data:
+    session_secret: ${SECRET}
+
+- apiVersion: apps/v1beta1
+  kind: Deployment
+  metadata:
+    name: autoheal
+    labels:
+      app: autoheal
+  spec:
+    selector:
+      matchLabels:
+        app: autoheal
+    replicas: 1
+    template:
+      metadata:
+        labels:
+          app: autoheal
+      spec:
+        serviceAccountName: autoheal
+        volumes:
+        - name: config
+          secret:
+            secretName: autoheal-config
+        - name: proxy-tls
+          secret:
+            secretName: autoheal-proxy-tls
+        - name: proxy-cookie
+          secret:
+            secretName: autoheal-proxy-cookie
+        containers:
+        - name: proxy
+          image: openshift/oauth-proxy:v1.1.0
+          imagePullPolicy: IfNotPresent
+          volumeMounts:
+          - mountPath: /etc/tls/private
+            name: proxy-tls
+          - mountPath: /etc/proxy/secrets
+            name: proxy-cookie
+          ports:
+          - containerPort: 8443
+            name: public
+          args:
+          - --https-address=:8443
+          - --provider=openshift
+          - --openshift-service-account=autoheal
+          - --upstream=http://localhost:9099
+          - --tls-cert=/etc/tls/private/tls.crt
+          - -email-domain=*
+          - '-openshift-sar={ "resource": "secrets", "verb": "get", "name": "autoheal-access-key", "namespace": "openshift-autoheal" }'
+          - '-openshift-delegate-urls={ "/": { "resource": "secrets", "verb": "get", "name": "autoheal-access-key", "namespace": "openshift-autoheal" } }'
+          - -tls-key=/etc/tls/private/tls.key
+          - -client-secret-file=/var/run/secrets/kubernetes.io/serviceaccount/token
+          - -cookie-secret-file=/etc/proxy/secrets/session_secret
+          - -openshift-ca=/etc/pki/tls/cert.pem
+          - -openshift-ca=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+        - name: receiver
+          image: ${IMAGE}
+          imagePullPolicy: IfNotPresent
+          volumeMounts:
+          - name: config
+            mountPath: /etc/autoheal/config.d
+          command:
+          - /usr/bin/autoheal
+          args:
+          - server
+          - --config-file=/etc/autoheal/config.d
+          - --logtostderr
+
+- apiVersion: v1
+  kind: Service
+  metadata:
+    name: receiver
+    labels:
+      app: autoheal
+    annotations:
+      service.alpha.openshift.io/serving-cert-secret-name: autoheal-proxy-tls
+  spec:
+    selector:
+      app: autoheal
+    ports:
+    - name: autoheal
+      port: 443
+      targetPort: 8443
```

Comment:
I have been using named port references in Service definitions (`public` in this case)... Seems more resilient to change. Just a random observation for future consideration

Reply:
I am open to do that. Can you elaborate? Maybe point me to an example?

Reply:
Sure... https://kubernetes.io/docs/concepts/services-networking/service/#defining-a-service

>Note that a Service can map an incoming port to any targetPort. By default the targetPort will be set to the same value as the port field. Perhaps more interesting is that targetPort can be a string, referring to the name of a port in the backend Pods. The actual port number assigned to that name can be different in each backend Pod. This offers a lot of flexibility for deploying and evolving your Services. For example, you can change the port number that pods expose in the next version of your backend software, without breaking clients.

Reply:
Thanks @ironcladlou! @elad661 can you take care of this?